test/Lachain.CoreTest/RPC/HTTP/Web3/AccountServiceWeb3Test.cs

```diff
@@ -136,30 +150,74 @@ public void Test_GetTransactionCount()
+        // Changed GetTransactionCount to public
+        public void Test_GetTransactionCount_pending()
+        {
+
+            var rawTx2 = "0xf8848001832e1a3094010000000000000000000000000000000000000080a4c76d99bd000000000000000000000000000000000000000000042300c0d3ae6a03a0000075a0f5e9683653d203dc22397b6c9e1e39adf8f6f5ad68c593ba0bb6c35c9cd4dbb8a0247a8b0618930c5c4abe178cbafb69c6d3ed62cfa6fa33f5c8c8147d096b0aa0";
+            var ethTx = new TransactionChainId(rawTx2.HexToBytes());
+            var address = ethTx.Key.GetPublicAddress().HexToBytes().ToUInt160();
+
+            var txCountBefore = _apiService!.GetTransactionCount(ethTx.Key.GetPublicAddress(), "pending");
+            Assert.AreEqual(txCountBefore, 0);
+
+            TransactionUtils.SetChainId(41);
+            Execute_dummy_transaction(false);
+
+            var txCountAfter = _apiService!.GetTransactionCount(ethTx.Key.GetPublicAddress(), "pending");
+            Assert.AreEqual(txCountAfter, 1);
+        }
+
         [Test]
         // Changed GetCode to public
-        public void Test_GetCode()
+        public void Test_GetCode_latest()
         {
             var address = "0x6bc32575acb8754886dc283c2c8ac54b1bd93195";
             var adCode = _apiService!.GetCode(address, "latest");
             Assert.AreEqual(adCode, "");
         }
 
+        [Test]
+        // Changed GetCode to public
+        public void Test_GetCode_pending()
+        {
+            var keyPair = _privateWallet.EcdsaKeyPair;
+            var address = "0x9210567c1f79e9e9c3634331158d3143e572c001";
+
+            var chainId = _configManager.GetConfig<NetworkConfig>("network")?.ChainId;
+            TransactionUtils.SetChainId((int)chainId!);
+
+            // Deploy contract 
+            var byteCode = ByteCodeHex.HexToBytes();
+            //Assert.That(VirtualMachine.VerifyContract(byteCode, true), "Unable to validate smart-contract code");
+            var from = keyPair.PublicKey.GetAddress();
+            var nonce = _stateManager.LastApprovedSnapshot.Transactions.GetTotalTransactionCount(from);
+            var contractHash = from.ToBytes().Concat(nonce.ToBytes()).Ripemd();
+            var tx = _transactionBuilder.DeployTransaction(from, byteCode);
+
+            var signedTx = Signer.Sign(tx, keyPair);
+            var result = _transactionPool.Add(signedTx);
+
+            var adCode = _apiService!.GetCode(address, "pending");
+            var adCode_bytes = adCode.HexToBytes();
+            Assert.AreEqual(adCode_bytes, byteCode);
+        }
+
         // Below methods Execute a Transaction
-        private void Execute_dummy_transaction()
+        private void Execute_dummy_transaction(bool generateblock, string rawTx = "0xf8848001832e1a3094010000000000000000000000000000000000000080a4c76d99bd000000000000000000000000000000000000000000042300c0d3ae6a03a0000075a0f5e9683653d203dc22397b6c9e1e39adf8f6f5ad68c593ba0bb6c35c9cd4dbb8a0247a8b0618930c5c4abe178cbafb69c6d3ed62cfa6fa33f5c8c8147d096b0aa0")
```

Comment:
`rawTx` - we are using the constant value here, for now, in that case, it would be good to move that paramere inside the function or better to initialise it at the class level